### PR TITLE
chore: add sonic to llamalend

### PR DESCRIPTION
**Pull request by OnlyJousting**

### CODE REVIEW

File: apps/main/src/lend/networks.ts [hunk 1]
```diff
@@ -31,6 +31,11 @@ const networksConfig = {
     showInSelectNetwork: true,
     pricesData: true,
   },
+  146: {
```
Comment (DanielSchiavini) [inline apps/main/src/lend/networks.ts:34]:
```suggestion
  [Chain.Sonic]: {
```

**Review by DanielSchiavini — DISMISSED** [review]

**Review by DanielSchiavini — DISMISSED** [review]

File: apps/main/src/lend/entities/chain/chain-query.ts [hunk 2]
```diff
@@ -6,9 +6,11 @@ import { ChainId } from '@/lend/types/lend.types'
 
 export const { useQuery: useOneWayMarketNames, prefetchQuery: prefetchMarkets } = queryFactory({
   queryKey: ({ chainId }: ChainParams) => ['chain', { chainId }, 'markets'] as const,
-  queryFn: async (_: ChainQuery<ChainId>): Promise<string[]> => {
+  queryFn: async (chainId: ChainQuery<ChainId>): Promise<string[]> => {
+    const useAPI = chainId.chainId === 146 ? false : true // disable API for sonic
```
Comment (DanielSchiavini) [inline apps/main/src/lend/entities/chain/chain-query.ts:10]:
```suggestion
    const useAPI = chainId.chainId !== 146 // disable API for sonic
```

**Review by DanielSchiavini — APPROVED** [review]

File: apps/main/src/lend/networks.ts [hunk 3]
```diff
@@ -1,3 +1,4 @@
+import { Chain } from 'curve-ui-kit/src/utils'
```
Comment (0xAlunara) [inline apps/main/src/lend/networks.ts:1]:
```suggestion
import type { Chain } from 'curve-ui-kit/src/utils'
```

**Review by 0xAlunara — DISMISSED** [review]

File: apps/main/src/lend/networks.ts (hunk 3 above)
Comment (0xAlunara) [reply apps/main/src/lend/networks.ts:1]: oh lol mistakes were made in that case haha

File: apps/main/src/lend/lib/apiLending.ts [hunk 4]
```diff
@@ -497,9 +498,22 @@ const user = {
...
-            market.userLoss(),
           ])
 
+        // Fetch user loss separately to prevent prices-api dependency from blocking contract read data
+        let loss: UserLoss
+        try {
+          loss = await market.userLoss()
+        } catch (error) {
+          console.error('Failed to fetch user loss:', error)
+          loss = {
```
Comment (0xAlunara) [inline apps/main/src/lend/lib/apiLending.ts:509]: could put this initialization above the `try` i guess

**Review by 0xAlunara — DISMISSED** [review]

File: apps/main/src/lend/lib/apiLending.ts (hunk 4 above)
Comment (DanielSchiavini) [reply apps/main/src/lend/lib/apiLending.ts:509]: Please let's stop using 0 for fallback :sob: 
what f the user has some loss and we report 0? It's just wrong!
Any place using these values might want to handle it differently

File: apps/main/src/lend/entities/chain/chain-query.ts [hunk 5]
```diff
@@ -6,9 +6,11 @@ import { ChainId } from '@/lend/types/lend.types'
 
 export const { useQuery: useOneWayMarketNames, prefetchQuery: prefetchMarkets } = queryFactory({
   queryKey: ({ chainId }: ChainParams) => ['chain', { chainId }, 'markets'] as const,
-  queryFn: async (_: ChainQuery<ChainId>): Promise<string[]> => {
+  queryFn: async (chainId: ChainQuery<ChainId>): Promise<string[]> => {
+    const useAPI = chainId.chainId !== 146 // disable API for sonic
```
Comment (DanielSchiavini) [inline apps/main/src/lend/entities/chain/chain-query.ts:10]:
```suggestion
    const useAPI = true
```

File: apps/main/src/lend/lib/apiLending.ts (hunk 4 above)
Comment (0xAlunara) [reply apps/main/src/lend/lib/apiLending.ts]: On error loss remains `undefined` now, and in those cases a '?' should be shown

File: apps/main/src/lend/entities/chain/chain-query.ts (hunk 5 above)
Comment (0xAlunara) [reply apps/main/src/lend/entities/chain/chain-query.ts:10]: Turns out this cannot yet be done, as gauges don't seem to work yet for the API

**Review by 0xAlunara — DISMISSED** [review]

**Review by DanielSchiavini — DISMISSED** [review]

**Review by 0xAlunara — APPROVED** [review]

**Review by pilotvietnam — APPROVED** [review]
